File: zerojudge/f634/main.py
table = [[set() for _ in range(3)]]
navy, army, air = 0
officer, sergeant, soldier = 0
tm=0
N, M = map(int, input().split())
for i in range(M):
    name, A, B = map(str, input().split())
    A = int(A)-1
    B = int(B)-1
    if (name in table[A][B]): continue
    else:
        table[A][B].add(name)
        if (A == 0): navy+=1
        elif (A == 1): army+=1
        else: air+=1
        if (B == 0): officer+=1
        elif (B == 1): sergeant+=1
        else: soldier+=1
        tm+=1

print(f"navy:{navy} army:{army} air:{air}")
print(f"officer:{officer} sergeant:{sergeant} soldier:{soldier}")
print(f"survival rate:{round(tm/N*1000)/10}%")

File: zerojudge/f634/main.cpp
#include<bits/stdc++.h>
using namespace std;

signed main(){
	ios::sync_with_stdio(false);
	cin.tie(0);
	int N, M, A, B;
	string name;
	int navy=0, army=0, air=0, officer=0, sergeant=0, soldier=0;
	int vec_m=0, tm=0;
	cin>>N>>M;
	vector<vector<set<string> > > table(3, vector<set<string> >(3, set<string>()));
	for (int m; m<M; m++){
		cin>>name>>A>>B;
		A--;
		B--;
		if (table[A][B].count(name)) continue;
		else{
			table[A][B].insert(name);
			if (A==0) navy++;
			else if (A==1) army++;
			else air++;
			if (B==0) officer++;
			else if (B==1) sergeant++;
			else soldier++;
			tm++;
		}
	}
	
	cout<<"navy:"<<navy<<" army:"<<army<<" air:"<<air<<"\n";
	cout<<"officer:"<<officer<<" sergeant:"<<sergeant<<" soldier:"<<soldier<<"\n";
	cout<<fixed<<setprecision(1);
	cout<<"survival rate: "<<(float)tm/N*100<<"%";
	return 0;
}